examples/contactnets_simple.py

```diff
@@ -225,9 +225,15 @@ def regenerate_callback(epoch: int, learned_system: System,
         cast(MultibodyLearnableSystem, learned_system).generate_updated_urdfs()
 
     # Trains system.
+    print("Training.")
     experiment.train(
         regenerate_callback if regenerate else default_epoch_callback)
 
+    # Store the results of the trained system.
```

Comment:
The intended usage of `get_results` is that you don't have to call `train` first -- it calls `train` for you.

Reply:
After some discussion, the workable solution we came to is the following:

- `get_results` is renamed `generate_results` to indicate more clearly that it will start/continue the training process as necessary.
- The newly named `generate_results` now returns both the learned model corresponding to the best validation loss and the final evaluation statistics.
- The example script `contactnets_simple.py` calls `generate_results` only, not also `train`.

These changes are implemented in the latest commit in this PR.